src/util/adb.js

```diff
@@ -210,10 +210,46 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, remove?: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    let directories = false;
+
+    await wrapADBCall(async () => {
+      log.debug('Finding older artifacts');
+      const files = await adbClient.readdir(deviceId, '/sdcard');
+
+      for (const file of files) {
+        if (!file.isDirectory() ||
+              !file.name.startsWith('web-ext-artifacts-')) {
+          continue;
+        }
+        directories = true;
+
+        if (!remove) {
+          break;
+        }
+
+        const temp = '/sdcard/';
+        const artifactsDir = temp + file.name;
```

Comment:
As I mentioned in a previous review comment, we should use template strings instead of composing strings with the `+` operator (to be even more clear about it this time: something like `temp + file.name` should become `${temp}${file.name}` enclosed inside two ` characters).

Besides that lets define a `const DEVICE_DIR_BASE = '/sdcard/'` and a `const ARTIFACTS_DIR_PREFIX = 'web-ext-artifacts'` at the start of this file (e.g. right before the `const log = ...`)
and then use those const in both `getOrCreateArtifactsDir` and `checkOrCleanArtifacts` to define the string paths we need.